pkg/kubelet/cm/topologymanager/numatopology/fakesysfs/fake.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fakesysfs
+
+import "fmt"
+
+type FakeSysFs struct {
+	distances    map[int]string
+	distancesErr error
+}
+
+func (fs *FakeSysFs) GetDistances(nodeId int) (string, error) {
+	if fs.distancesErr != nil {
+		return "", fs.distancesErr
+	}
+
+	if _, ok := fs.distances[nodeId]; !ok {
+		return "", fmt.Errorf("distance not found")
+	}
+
+	return fs.distances[nodeId], nil
+}
+
+func (fs *FakeSysFs) SetDistances(nodeId int, distances string, err error) {
+	if fs.distances == nil {
+		fs.distances = map[int]string{nodeId: distances}
+	} else {
+		fs.distances[nodeId] = distances
+	}
+	fs.distancesErr = err
+}
```

Comment:
I don't think we should introduce a whole new subpackage like this just for testing purposes. 
In general, I think we can simplify your whole approach of FakeSysFs and RealSysfs into a smaller, simpler abstraction. Let me add my thoughts in a separate comment after finishing this review pass.

Reply:
This subpackage will be removed when new release of Cadvisor will be available(the code to expose Distances is already merged), so I just tried to do it similliar how it is done in Cadvisor.